hermes/crates/cbork/cddl-parser/src/lib.rs

```diff
@@ -45,35 +102,50 @@ pub const POSTLUDE: &str = include_str!("grammar/postlude.cddl");
 /// # Examples
 ///
 /// ```rs
-/// use cddl_parser::parse_cddl;
+/// use cddl_parser::{parse_cddl, Extension};
 /// use std:fs;
 ///
 /// let input = fs::read_to_string("path/to/your/file.cddl").unwrap();
-/// let result = parse_cddl(&input);
+/// let result = parse_cddl(&input, &Extension::CDDLParser);
 /// assert!(result.is_ok());
 /// ```
-pub fn parse_cddl(input: &str) -> Result<(), Box<CDDLError>> {
-    let result = CDDLParser::parse(Rule::cddl, input);
-
-    match result {
-        Ok(c) => println!("{c:?}"),
-        Err(e) => {
-            println!("{e:?}");
-            println!("{e}");
-            return Err(Box::new(CDDLError::from(e)));
+pub fn parse_cddl(input: &str, extension: &Extension) -> Result<(), Box<CDDLError>> {
+    let result = match extension {
+        Extension::RFC8610Parser => {
+            rfc_8610::RFC8610Parser::parse(rfc_8610::Rule::cddl, input)
+                .map(|_| ())
+                .map_err(CDDLErrorType::RFC8610)
         },
-    }
+        Extension::RFC9615Parser => {
+            rfc_9615::RFC8610Parser::parse(rfc_9615::Rule::cddl, input)
+                .map(|_| ())
+                .map_err(CDDLErrorType::RFC9615)
+        },
+        Extension::CDDLParser => {
+            cddl::RFC8610Parser::parse(cddl::Rule::cddl, input)
+                .map(|_| ())
+                .map_err(CDDLErrorType::CDDL)
+        },
+    };
 
-    Ok(())
+    result.map_err(|e| {
+        println!("{e:?}");
+        println!("{e}");
+
+        Box::new(CDDLError::from(e))
+    })
 }
 
 #[cfg(test)]
 mod tests {
-    use crate::{parse_cddl, POSTLUDE};
+    use crate::*;
+
+    // CDDL Standard Postlude - read from an external file
+    const POSTLUDE: &str = include_str!("grammar/postlude.cddl");
```

Comment:
I think we should promote this outside the test module, up at the top of the file after the parser modules are defined.
I also think in the parser now, we should get `input` and append the `POSTLUDE` to it in `parse_cddl`.  This is because every CDDL file should have the `POSTLUDE` added to it according to the spec.

This test then can just try and parse an empty string which is equivalent to just parsing the `POSTLUDE`